content/master/concepts/compositions.md

```diff
@@ -0,0 +1,993 @@
+Composition template, the Composite Resource creates two new managed resources
+with all the provided 
+{{<hover label="resources" line="17">}}spec.forProvider{{</hover>}} settings.
+
+The {{<hover label="resources" line="17">}}spec.forProvider{{</hover>}}
+supports any settings used in a managed resource including
+applying `annotations` and `labels` or using a specific 
+`providerConfigRef`.
+
+{{<hint "note" >}}
+Compositions allow applying a `metadata.name` but ignores it. The
+`metadata.name` field doesn't influence the name
+of the managed resource in Crossplane or the external resource inside the 
+Provider.
+
+Use the `crossplane.io/external-name` annotation on the resource to influence
+the external resource name.
+{{< /hint >}}
+
+#### Template a ProviderConfig
+
+Compositions can define a ProviderConfig like it defines managed resources.
+Generating a ProviderConfig may be useful in providing unique credentials to
+each deployment.
+
+
+```yaml {copy-lines="none"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+spec:
+  resources:
+    - name: my-aws-provider-config
+      base:
+        apiVersion: aws.upbound.io/v1beta1
+        kind: ProviderConfig
+        spec:
+          source: Secret
+          secretRef:
+            namespace: crossplane-system
+            name: aws-secret
+            key: creds
+```
+
+
+#### Template another Composite Resource
+
+Compositions can use existing Composite Resources to define more complicated
+templates. 
+
+A common use case is a Composition that uses other
+Compositions. For example, creating a Composition to create a standard set of 
+networking resources that other Compositions reference. 
+
+This example networking Composition defines the set of resources required to 
+create a new AWS virtual network. This includes a 
+{{<hover label="xnetwork" line="8">}}VPC{{</hover>}},
+{{<hover label="xnetwork" line="13">}}InternetGateway{{</hover>}} and
+{{<hover label="xnetwork" line="18">}}Subnet{{</hover>}}.
+
+```yaml {copy-lines="none",label="xnetwork"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+spec:
+  resources:
+    - name: vpc-resource
+      base:
+        apiVersion: ec2.aws.upbound.io/v1beta1
+        kind: VPC
+        # Removed for Brevity
+    - name: gateway-resource
+      base:
+        apiVersion: ec2.aws.upbound.io/v1beta1
+        kind: InternetGateway
+        # Removed for Brevity
+    - name: subnet-resource
+      base:
+        apiVersion: ec2.aws.upbound.io/v1beta1
+        kind: Subnet
+        # Removed for Brevity
+  compositeTypeRef:
+    apiVersion: aws.platformref.upbound.io/v1alpha1
+    kind: XNetwork
+```
+
+The {{<hover label="xnetwork" line="20">}}compositeTypeRef{{</hover>}} gives
+this Composition a 
+{{<hover label="xnetwork" line="21">}}apiVersion{{</hover>}} and
+{{<hover label="xnetwork" line="22">}}kind{{</hover>}} to reference in another
+Composition. 
```

Comment:
This is an interesting way to frame it. Not wrong, though. Maybe clarify that it specifies a type of XR another Composition can reference?